app/views/proposals/_tooltip.html.haml

```diff
@@ -1,11 +1,12 @@
 - progress_status = event.progress_status
 %ul.list-unstyled
-  %li{'class'=>class_for_todo(progress_status['registered'])}
-    %span{'class'=>icon_for_todo(progress_status['registered'])}
-      - if progress_status['registered']
-        Speaker(s) registered to the conference
-      - else
-        = link_to 'Speaker(s) not registered to the conference', new_conference_conference_registration_path(event.program.conference.short_title)
+  - if @conference.registration_open? || (@conference.user_registered? current_user)
```

Comment:
Instead of only checking those 2 conditions here, it's best that we check if the user can create a new registration. 

**How do we know if the user is allowed create a new registration?**
We are using [CanCanCan](https://github.com/CanCanCommunity/cancancan/wiki) to define what a user can or cannot do (which action of which model the user can access). So let's have CanCanCan do its job - ability.rb file contains the necessary definitions of what a user is allowed, or not, to do.

**Why we should do that?**
Because instead of checking against specific conditions, we can check what we really care about, which is whether or not the user is allowed to perform a certain action. 

Your code is  now only checking for 2 conditions

 - if the registration is open and
 - if the user is registered

There could be more conditions in the future (in fact there already are https://github.com/openSUSE/osem/blob/master/app/models/ability.rb#L69, but let's not focus on that right now). 

Checking if the user can execute an action will always work properly, without requiring any modifications to this code here, even if there are more conditions than those two.

Reply:
@namangupta01 

> If we follow above four statements, then we cannot use the ability of user to create new registration.

Why not?

Reply:
@differentreality I misunderstood something. I am correcting this. Thanks!

Reply:
@differentreality This issue is dependent on the issue #1628 , which fixes the ability of user to create registration. Please have a look at the corresponding PR.

Reply:
Regardless of the other PR, do you understand my point about checking if user can create a new registration?

Reply:
Yes i understood that we have to show the link and encouragement text to a user only when a user can register, if a user is not allowed to do registration or the user is already registered then we doesn't have to show them the encouragement text and tooltip link and also doesn't count the registration in progress bar as well. Is this the correct way i am thinking @differentreality ?

Reply:
Yes, exactly. 

So instead of the if condition you are using on line 3, you should check for the user's ability to register.

Reply:
okay :+1: Should i correct my pr or wait for issue #1628 because without this my changes will not show correct output.

Reply:
You can work on the changes, as we wait for #1628 to be merged.

Reply:
sure :+1:

Reply:
Ping

Reply:
@differentreality respective changes are done! Have a look!